File: frontend/src/api/batch.ts
import client from './client'

export interface BatchUpdateRequest {
  ids: string[]
  rating?: number
  is_favorite?: boolean
  needs_improvement?: boolean
  add_tags?: string[]
  remove_tags?: string[]
}

export interface BatchDeleteRequest {
  ids: string[]
  permanent?: boolean
}

export interface BatchRestoreRequest {
  ids: string[]
}

export const batchApi = {
  update: async (data: BatchUpdateRequest): Promise<void> => {
    await client.post('/bulk/update', data)
  },

  delete: async (data: BatchDeleteRequest): Promise<void> => {
    await client.post('/bulk/delete', data)
  },

  restore: async (data: BatchRestoreRequest): Promise<void> => {
    await client.post('/bulk/restore', data)
  },
}

File: backend-rs/src/dto/batch.rs
//! Bulk operation request DTOs (mirror endpoints/batch.py).

use uuid::Uuid;

#[derive(Debug, serde::Deserialize)]
pub struct BatchUpdateRequest {
    pub ids: Vec<Uuid>,
    pub rating: Option<i32>,
    pub is_favorite: Option<bool>,
    pub needs_improvement: Option<bool>,
    pub add_tags: Option<Vec<String>>,
    pub remove_tags: Option<Vec<String>>,
}

#[derive(Debug, serde::Deserialize)]
pub struct BatchDeleteRequest {
    pub ids: Vec<Uuid>,
    #[serde(default)]
    pub permanent: bool,
}

#[derive(Debug, serde::Deserialize)]
pub struct BatchRestoreRequest {
    pub ids: Vec<Uuid>,
}
